src/serac/infrastructure/about.cpp

```diff
@@ -0,0 +1,216 @@
+
+  // Conduit
+#ifdef SERAC_USE_CONDUIT
+  about += format("Conduit Version: {0}\n", CONDUIT_VERSION);
+#else
+  disabled_libs.push_back("Conduit");
+#endif
+
+  // HDF5
+#ifdef SERAC_USE_HDF5
+  unsigned int h5_maj, h5_min, h5_rel;
+  std::string  h5_version;
+  if (H5get_libversion(&h5_maj, &h5_min, &h5_rel) < 0) {
+    SLIC_ERROR("Failed to retrieve HDF5 version.");
+  } else {
+    h5_version = format("{0}.{1}.{2}", h5_maj, h5_min, h5_rel);
+  }
+  about += format("HDF5 Version:    {0}\n", h5_version);
+#else
+  disabled_libs.push_back("HDF5");
+#endif
+
+  // Lua
+#ifdef SERAC_USE_LUA
+  std::string lua_version{LUA_RELEASE};
+  if (axom::utilities::string::startsWith(lua_version, "Lua ")) {
+    lua_version.erase(0, 4);
+  }
+  about += format("Lua Version:     {0}\n", lua_version);
+#else
+  disabled_libs.push_back("Lua");
+#endif
+
+  // MFEM
+  const char* mfem_version = mfem::GetVersionStr();
+  if (mfem_version == nullptr) {
+    SLIC_ERROR("Failed to retrieve MFEM version.");
+  }
+  const char* mfem_sha = mfem::GetGitStr();
+  if (mfem_sha == nullptr) {
+    SLIC_ERROR("Failed to retrieve MFEM Git SHA.");
+  }
+  std::string mfem_full_version = std::string(mfem_version);
+  if (axom::utilities::string::startsWith(mfem_full_version, "MFEM ")) {
+    mfem_full_version.erase(0, 5);
+  }
+  if (mfem_sha[0] != '\0') {
+    mfem_full_version += format(" (Git SHA: {0})", mfem_sha);
+  }
+  about += format("MFEM Version:    {0}\n", mfem_full_version);
+
+  // RAJA
+#ifdef SERAC_USE_RAJA
+  about += format("RAJA Version:    {0}.{1}.{2}\n", RAJA_VERSION_MAJOR, RAJA_VERSION_MINOR, RAJA_VERSION_PATCHLEVEL);
+#else
+  disabled_libs.push_back("RAJA");
+#endif
+
+  // Tribol
+#ifdef SERAC_USE_TRIBOL
+  // TODO: This can be un-hardcoded when Tribol MR!59 gets merged and we update to it
+  //  about += format("Tribol Version:    {0}\n", TRIBOL_VERSION_FULL);
+  about += "Tribol Version:    0.1.0\n";
+#else
+  disabled_libs.push_back("Tribol");
+#endif
+
+  // Umpire
+#ifdef SERAC_USE_UMPIRE
+  about += format("Umpire Version:  {0}.{1}.{2}\n", umpire::get_major_version(), umpire::get_minor_version(),
+                  umpire::get_patch_version());
+#else
+  disabled_libs.push_back("Umpire");
+#endif
+
+  about += "\n";
+
+  about += "Disabled Libraries:\n";
+  if (disabled_libs.size() == 0) {
+    about += "None\n";
+  } else {
+    for (auto& lib : disabled_libs) {
+      about += lib + "\n";
+    }
+  }
+
+  return about;
+}
+
+std::string gitSHA()
+{
+  // Note: This is generated at configure time so that
+  // when the sha changes it doesn't force a full rebuild
+  // of Serac. Do not add it to the config header.
+  // Note: This will not update unless you re-run CMake between commits.
+  return SERAC_GIT_SHA;
+}
+
+std::string version(bool add_SHA)
+{
+  std::string version =
+      axom::fmt::format("v{0}.{1}.{2}", SERAC_VERSION_MAJOR, SERAC_VERSION_MINOR, SERAC_VERSION_PATCH);
+
+  std::string sha = gitSHA();
+  if (add_SHA && !sha.empty()) {
```

Comment:
could `gitSHA()` ever return an empty string?

Reply:
It will return an empty string when we do a github release tarball because it will not contain a `.git` folder

Reply:
good point, I was not thinking about that!